content/blog/vue-performance.md

```diff
@@ -0,0 +1,86 @@
+---
+author: "Alex Hill"
+date: 2020-03-01
+title: Handling long arrays performantly in Vuejs
+tags:
+ - Vue
+ - Javascript
+ - performance
+---
+
+
+## TLDR;
+Use `Object.freeze` on large objects that don't have to be reactive. If you want to know a bit more about why this works 
+and how big a performance gain it is, read on.
+
+## Context
+
+Reactive JavaScript frameworks like Vuejs are amazing for developing Single Page Apps like the one in our project 
+[Naomi](/projects/#naomi). Everything is connected as if by magic - when data changes, the UI updates. But that magic 
+has overheads associated with it that need to be understood to optimise for performance where large-ish datasets are 
+being handled. We found when using Vue naively that our app couldn't handle a 10MB dataset performantly, but 
+understanding how reactivity works in Vue helped us to find the fix.
+
+Naomi provides visualisations of HIV epidemic indicators 
+across a country, at a fine-grained regional level. Users can filter the data by many parameters, 
+e.g. age, sex, year, level of of detail (from the whole country down to individual cities.)
+
+<img src="/img/choropeek.gif" alt="Gif of the map widget in Naomi" />
+
+The front-end app is written in [Vue.js](https://vuejs.org/) + [Vuex](https://vuex.vuejs.org/) and for the map plots we 
+are using the excellent [Leaflet](https://leafletjs.com/) library. The HIV indicator data and GeoJSON are returned from
+ an API and are in the  order of 10MB in size; the fitering of data happens in the front-end and the map is redrawn each 
```

Comment:
"and are _of_ the order of 10..." sounds more natural to my ears but I'm doubting it as I write this...